src/hooks/useSwapCallback.ts

```diff
@@ -146,6 +146,21 @@ export function useSwapCallback(
               web3Provider.provider.isMetaMask = isMetamask
             }
 
+            const parsed = ethers.utils.parseTransaction(signedTx)
+
+            if (parsed.from !== account) {
+              // console.log("DETECTED ACCOUNTS DON'T MATCH")
+              if (isMetamask) {
+                throw new Error(
+                  'MistX does not support Hard Wallets connected through MetaMask. If you are using Ledger, please connect it directly.'
+                )
+              } else {
+                throw new Error(
+                  'The wallet used does not support eth_signTransaction. Try using MetaMask or connect your Ledger directly.'
```

Comment:
This would be non-ledger users, suggested copy:

`'The wallet used does not support our signing method. Please use a supported wallet such as MetaMask.'`